packages/backend/src/core/discovery/proxies/auto/Eip2535Proxy.ts

```diff
@@ -0,0 +1,33 @@
+// EIP-2535: Diamonds, Multi-Facet Proxy
+// https://eips.ethereum.org/EIPS/eip-2535#a-note-on-implementing-interfaces
+// every contract implementing this standard needs to have facetAddresses() view function
+
+import { EthereumAddress } from '@l2beat/types'
+
+import { DiscoveryProvider } from '../../provider/DiscoveryProvider'
+import { getCallResult } from '../../utils/getCallResult'
+import { ProxyDetection } from '../types'
+
+export async function detectEip2535proxy(
+  provider: DiscoveryProvider,
+  address: EthereumAddress,
+): Promise<ProxyDetection | undefined> {
+  const facets = await getCallResult<EthereumAddress[]>(
+    provider,
+    address,
+    'function facetAddresses() external view returns (address[] memory facetAd)',
```

Comment:
This will only work for diamonds with `IDiamondLoupe`
https://eips.ethereum.org/EIPS/eip-2535#inspecting-facets--functions

We don't know if a diamond has this so we should rely on the standard events instead:
https://eips.ethereum.org/EIPS/eip-2535#addingreplacingremoving-functions

Reply:
I agree this is different interface, but the EIP-2535 specification states: `Diamonds must support inspecting facets and functions by implementing the IDiamondLoupe interface`. Can't we assume that all Diamonds will implement this interface?

Reply:
@sz-piotr

Reply:
I missed that completely when reading the standard. I think that's fine then. We'll only revisit this if we encounter future exceptions :)